auth/ante.go

```diff
@@ -157,16 +176,45 @@ func processConfirmSig(
 
 // Checks that utxo at the position specified exists, matches the address in the SpendMsg
 // and returns the denomination associated with the utxo
-func checkUTXO(ctx sdk.Context, mapper utxo.Mapper, position types.PlasmaPosition, addr common.Address) sdk.Result {
+func checkUTXO(ctx sdk.Context, plasmaClient *eth.Plasma, mapper utxo.Mapper, position types.PlasmaPosition, addr common.Address) sdk.Result {
+	var inputAddress []byte
 	input := mapper.GetUTXO(ctx, addr.Bytes(), &position)
-	if !input.Valid {
-		return sdk.ErrUnknownRequest(fmt.Sprintf("UTXO trying to be spent, is not valid: %v.", position)).Result()
+	if position.IsDeposit() && reflect.DeepEqual(input, utxo.UTXO{}) {
+		deposit, ok := DepositExists(position.DepositNum, plasmaClient)
+		if !ok {
+			return utxo.ErrInvalidUTXO(2, "Deposit UTXO does not exist yet").Result()
+		}
+		inputAddress = deposit.Owner.Bytes()
+	} else {
+		if !input.Valid {
+			return sdk.ErrUnknownRequest(fmt.Sprintf("UTXO trying to be spent, is not valid: %v.", position)).Result()
+		}
 	}
 
 	// Verify that utxo owner equals input address in the transaction
-	if !reflect.DeepEqual(input.Address, addr.Bytes()) {
-		return sdk.ErrUnauthorized(fmt.Sprintf("signer does not match utxo owner, signer: %X  owner: %X", addr.Bytes(), input.Address)).Result()
+	if !reflect.DeepEqual(inputAddress, addr.Bytes()) {
+		return sdk.ErrUnauthorized(fmt.Sprintf("signer does not match utxo owner, signer: %X  owner: %X", addr.Bytes(), inputAddress)).Result()
 	}
-
 	return sdk.Result{}
 }
+
+func DepositExists(nonce uint64, plasmaClient *eth.Plasma) (types.Deposit, bool) {
+	deposit, err := plasmaClient.GetDeposit(sdk.NewUint(nonce))
+
+	if err != nil {
+		return types.Deposit{}, false
+	}
+	return *deposit, true
+}
+
+func hasTXExited(plasmaClient *eth.Plasma, pos types.PlasmaPosition) sdk.Error {
```

Comment:
Should eventually check if inputs to position have exited. If they have then censor tx by failing on CheckTx. Uncensor once challenge is successful